app/services/flashes.js

```diff
@@ -87,11 +87,13 @@ export default Service.extend({
   removeFlash(msg) {
     setTimeout(() => {
       run(this, () => {
-        if (this.get('flashes.content')) {
-          return this.get('flashes.content').removeObject(msg);
+        if (this.get('flashes').length > 0) {
+          return this.get('flashes').removeObject(msg);
         }
       });
-    }, 15000);
+      // Fadeout is currently done separatly with css, and completes at 7s. Keeping the message around longer than that can result in weird situations
+      // where reloading a page can result in a message showing again that you thought was gone.
+    }, 7000);
```

Comment:
Thanks for aligning this! Maybe someday we'll refactor it in some kind of a more solid and reliable way 🤞